### DISCUSSION TITLE
Which coordinate system is used to calculate area of polygons?

### DISCUSSION THREAD

**Discussion by sairahul1526** [peterstace/simplefeatures#612] [Q&A]
as the title says

And can we change the default coordinate system?

**Comment by peterstace** [comment]
Simplefeatures doesn't know about coordinate systems, everything is just a plain number.

So, for an area calculation, the input units will flow through to become output units. For example, take this polygon of a park:

<img width="661" alt="Screenshot 2024-05-14 at 5 14 18 am" src="https://github.com/peterstace/simplefeatures/assets/5352655/6ec1caf6-14cf-4e3a-a200-e917d7c423a1">


```
package main

import (
	"fmt"
	"math"

	"github.com/peterstace/simplefeatures/geom"
)

func main() {
	g, err := geom.UnmarshalWKT("POLYGON((151.19536 -33.86561,151.19529 -33.86599,151.19621 -33.86624,151.19646 -33.86604,151.1964 -33.86575,151.19536 -33.86561))")
	if err != nil {
		panic(err)
	}
	fmt.Println("area (square degrees):", g.Area())
}
```

This will print:

```
area (square degrees): 4.939000000025871e-07
```

This will give you the area of the polygon _in square degrees_, which isn't very useful.

What you need to do is reproject the coordinates to a coordinate system that is [equal area](https://en.wikipedia.org/wiki/Equal-area_projection) and has more appropriate units. Any equal area will do, so I normally use a simple one such as [Lambert cylindrical equal area](https://en.wikipedia.org/wiki/Lambert_cylindrical_equal-area_projection).

This will give you the area that you're looking for:

```
package main

import (
	"fmt"
	"math"

	"github.com/peterstace/simplefeatures/geom"
)

func main() {
	g, err := geom.UnmarshalWKT("POLYGON((151.19536 -33.86561,151.19529 -33.86599,151.19621 -33.86624,151.19646 -33.86604,151.1964 -33.86575,151.19536 -33.86561))")
	if err != nil {
		panic(err)
	}
	fmt.Println("area (square meters):", g.Area(geom.WithTransform(wgs84ToLambertEqualArea)))
}

func wgs84ToLambertEqualArea(xy geom.XY) geom.XY {
	xy = xy.Scale(math.Pi / 180) // Convert from degrees to radians.
	xy.Y = math.Sin(xy.Y)        // Project onto the cylinder.
	const earthRadius = 6371000  // Scale to meters.
	return xy.Scale(earthRadius)
}
```

This should give you the true area in square meters:

```
area (square meters): 5070.689245845211
```
